app/components/SingleProduct.js

```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+
+import { login } from 'APP/app/reducers/auth'
+import { connect } from 'react-redux'
+
+const SingleProduct = (props) => (
+  <div>
+    <h2>{props.singleProduct.name}</h2>
```

Comment:
this is the right idea, but I think we should leave this out for now because it will throw errors! Instead, we should probably focus on making the action creators and dispatches work correctly first. Good work though!